plugin/nginx_plus_collectd.py

```diff
@@ -772,6 +834,42 @@ def get_processes(self):
         '''
         return self._send_get(self.processes_url)
 
+    def _get_api_version(self):
+        '''
+        Determines whether the Nginx-plus plugin has a versioned API or legacy API.
+        '''
+        newer_api_base_path = self.api_base_path if self.api_base_path is not None else '/api'
+        legacy_api_base_path = self.api_base_path if self.api_base_path is not None else '/status'
+        base_url = 'http://{}:{}'.format(self.status_host, str(self.status_port))
+
+        try:
+            response = requests.get("{}{}/{}".format(base_url, newer_api_base_path, DEFAULT_API_VERSION), auth=self.auth_tuple)
+            if response.status_code == requests.codes.ok:
```

Comment:
If the target api doesn't support 3 but does support 2 this flow is invalid.  Stepping back, why is a provided api version not being taken into consideration and only the default being checked?  What if 1 were provided?

Reply:
We have verified that we are getting all the metrics mentioned (default and non-default) in the [README.md](https://github.com/signalfx/collectd-nginx-plus/blob/master/README.md) using `APIVersion` of `1` in NGINX+ version r13 and r24.
Now, we are using the DEFAULT_API_VERSION of `1` while detecting the NGINX+ API type (versioned or legacy).